include/clangmetatool/collectors/references_data.h

```diff
@@ -0,0 +1,44 @@
+#ifndef INCLUDED_CLANGMETATOOL_COLLECTORS_REFERENCES_DATA_H
+#define INCLUDED_CLANGMETATOOL_COLLECTORS_REFERENCES_DATA_H
+
+#include <unordered_map>
+
+#include <clang/AST/Decl.h>
+
+#include <clangmetatool/types/file_uid.h>
+
+namespace clangmetatool {
+namespace collectors {
+
+/**
+ * The data collected by the References collector.
+ */
+class ReferencesData {
+public:
+    /**
+     * Store data for references
+     * and the context in which the references exist.
+     */
+    std::unordered_multimap<const clang::NamedDecl *, const clang::NamedDecl *> refs;
```

Comment:
Can we populate an inverted version of this map as well? I suspect traversing the graph both ways will be a very common use case.